pkgs/development/libraries/yder/fix-darwin.patch

```diff
@@ -0,0 +1,420 @@
+diff --git i/CMakeLists.txt w/CMakeLists.txt
+index 036ab73..c9c6869 100644
+--- i/CMakeLists.txt
++++ w/CMakeLists.txt
+@@ -87,33 +87,6 @@ else()
+   set(DISABLE_JOURNALD ON)
+ endif ()
+ 
+-# shared library
+-
+-add_library(yder SHARED ${LIB_SRC})
+-if (NOT MSVC)
+-    set_target_properties(yder PROPERTIES
+-            COMPILE_OPTIONS -Wextra
+-            PUBLIC_HEADER "${INC_DIR}/yder.h;${PROJECT_BINARY_DIR}/yder-cfg.h"
+-            VERSION "${LIBRARY_VERSION}"
+-            SOVERSION "${LIBRARY_SOVERSION}")
+-endif()
+-if (WIN32)
+-    set_target_properties(yder PROPERTIES SUFFIX "-${LIBRARY_VERSION_MAJOR}.dll")
+-endif ()
+-
+-target_link_libraries(yder ${LIBS} ${ORCANIA_LIBRARIES} ${SYSTEMD_LIBRARIES})
+-
+-# static library
+-
+-option(BUILD_STATIC "Build static library." OFF)
+-
+-if (BUILD_STATIC)
+-    add_library(yder_static STATIC ${LIB_SRC})
+-    target_compile_definitions(yder_static PUBLIC -DO_STATIC_LIBRARY)
+-    set_target_properties(yder_static PROPERTIES
+-            OUTPUT_NAME yder)
+-endif ()
+-
+ option (SEARCH_ORCANIA "Search for Orcania library" ON)
+ if (SEARCH_ORCANIA)
+     set(Orcania_FIND_QUIETLY ON) # force to find Orcania quietly
+@@ -145,6 +118,33 @@ else ()
+   set(PKGCONF_REQ_PRIVATE "liborcania")
+ endif ()
+ 
++# shared library
++
++add_library(yder SHARED ${LIB_SRC})
++if (NOT MSVC)
++    set_target_properties(yder PROPERTIES
++            COMPILE_OPTIONS -Wextra
++            PUBLIC_HEADER "${INC_DIR}/yder.h;${PROJECT_BINARY_DIR}/yder-cfg.h"
++            VERSION "${LIBRARY_VERSION}"
++            SOVERSION "${LIBRARY_SOVERSION}")
++endif()
++if (WIN32)
++    set_target_properties(yder PROPERTIES SUFFIX "-${LIBRARY_VERSION_MAJOR}.dll")
++endif ()
++
++target_link_libraries(yder ${LIBS} ${ORCANIA_LIBRARIES} ${SYSTEMD_LIBRARIES})
++
++# static library
++
++option(BUILD_STATIC "Build static library." OFF)
++
++if (BUILD_STATIC)
++    add_library(yder_static STATIC ${LIB_SRC})
++    target_compile_definitions(yder_static PUBLIC -DO_STATIC_LIBRARY)
++    set_target_properties(yder_static PROPERTIES
++            OUTPUT_NAME yder)
++endif ()
++
+ # build yder-cfg.h file
+ configure_file(${INC_DIR}/yder-cfg.h.in ${PROJECT_BINARY_DIR}/yder-cfg.h)
+ set (CMAKE_EXTRA_INCLUDE_FILES ${PROJECT_BINARY_DIR})
+diff --git i/Makefile w/Makefile
+deleted file mode 100644
```

Comment:
If understand it correctly from here on your patch just deletes a bunch of make files. Why not using `rm Makefile` etc in `postPatch`? That would be shorter and more clear.

Reply:
Oh, those deletions weren't supposed to be included - that was from some experimentation that I did. The main change is of that in `CMakeLists.txt`.